pkgs/os-specific/linux/wireguard/default.nix

```diff
@@ -1,73 +1,39 @@
-{ stdenv, fetchurl, libmnl, kernel ? null }:
+{ stdenv, fetchzip, kernel }:
 
 # module requires Linux >= 3.10 https://www.wireguard.io/install/#kernel-requirements
-assert kernel != null -> stdenv.lib.versionAtLeast kernel.version "3.10";
+assert stdenv.lib.versionAtLeast kernel.version "3.10";
 
-let
+stdenv.mkDerivation rec {
   name = "wireguard-${version}";
-
   version = "0.0.20180514";
 
-  src = fetchurl {
+  src = fetchzip {
```

Comment:
I don't think it is a good idea to split this file because now somebody might by accident update only one of the two packages. Why not mark the user space component as platform independent instead of duplicating the package?

Reply:
My understanding is that the versions of the module and tools are actually more-or-less independent, but I might be wrong.
I also find current logic with returning a different derivation base on the presence of the `kernel` argument somewhat, um, unorthodox.

> instead of duplicating the package
Which directory will we put it then? The kernel modules _really_ belongs to `os-specific/linux` and tools _really_ do not belong there. I considered having a separate `.nix` file with the source definition, but a) I was not sure which directory to put it into 🙂, b) see my previous point, I believe the versions are independent, so I chose the simplest path.

Reply:
This could as well be `inherit (wireguard-tools) src`. This shouldn't matter long-term, because Wireguard is going to be upstreamed to the kernel tree.

Reply:
Userspace and kernelspace need to be kept in think. If the netlink interface is extended they might become incompatible. This might change when wireguard gets upstream, but we are not there yet.